### PR TITLE
Enable wchar_t as string_view value_type

### LINKED ISSUE #193: Support wide char
Is it possible to support wide char ?

#define MAGIC_ENUM_USING_ALIAS_STRING_VIEW using string_view = std::wstring_view; 
#define MAGIC_ENUM_USING_ALIAS_STRING using string = std::wstring; 
#include "magic_enum.hpp"

wstring_view name = magic_enum::enum_name(Color::Red);

### PR DESCRIPTION
**Pull request by Neargye**

### CODE REVIEW

File: include/magic_enum.hpp [hunk 1]
```diff
@@ -244,17 +259,20 @@ class static_str {
...
+  constexpr const char_type* data() const noexcept { return chars_; }
 
   constexpr std::uint16_t size() const noexcept { return N; }
 
   constexpr operator string_view() const noexcept { return {data(), size()}; }
 
  private:
   template <std::uint16_t... I>
-  constexpr static_str(const char* str, std::integer_sequence<std::uint16_t, I...>) noexcept : chars_{str[I]..., '\0'} {}
+  constexpr static_str(const char* str, std::integer_sequence<std::uint16_t, I...>) noexcept : chars_{static_cast<char_type>(str[I])..., '\0'} {}
```
Comment (schaumb) [inline include/magic_enum.hpp:270]: static_cast<char_type>('\0');

File: include/magic_enum.hpp [hunk 2]
```diff
@@ -244,17 +259,20 @@ class static_str {
...
+
+  template <std::uint16_t... I>
+  constexpr static_str(string_view str, std::integer_sequence<std::uint16_t, I...>) noexcept : chars_{str[I]..., '\0'} {}
```
Comment (schaumb) [inline include/magic_enum.hpp:273]: static_cast<char_type>('\0');

File: include/magic_enum.hpp [hunk 3]
```diff
@@ -275,18 +293,18 @@ class static_str<0> {
 
 template <typename Op = std::equal_to<>>
 class case_insensitive {
-  static constexpr char to_lower(char c) noexcept {
-    return (c >= 'A' && c <= 'Z') ? static_cast<char>(c + ('a' - 'A')) : c;
+  static constexpr char_type to_lower(char_type c) noexcept {
+    return (c >= 'A' && c <= 'Z') ? static_cast<char_type>(c + ('a' - 'A')) : c;
```
Comment (schaumb) [inline include/magic_enum.hpp:297]: static_cast<char_type>('A') , static_cast<char_type>('Z')

File: include/magic_enum.hpp [hunk 4]
```diff
@@ -1225,7 +1243,7 @@ template <detail::enum_subtype S, typename E>
 // Returns name from enum-flags value.
 // If enum-flags value does not have name or value out of range, returns empty string.
 template <typename E>
-[[nodiscard]] auto enum_flags_name(E value, char sep = '|') -> detail::enable_if_t<E, string> {
+[[nodiscard]] auto enum_flags_name(E value, char_type sep = '|') -> detail::enable_if_t<E, string> {
```
Comment (schaumb) [inline include/magic_enum.hpp:1246]: static_cast<char_type>('|') 

Check this character too at static_assert check at line 141/142

File: include/magic_enum_containers.hpp [hunk 5]
```diff
@@ -546,7 +546,7 @@ class bitset {
     }
   }
 
-  constexpr explicit bitset(detail::raw_access_t, string_view sv, string_view::size_type pos = 0, string_view::size_type n = string_view::npos, char zero = '0', char one = '1')
+  constexpr explicit bitset(detail::raw_access_t, string_view sv, string_view::size_type pos = 0, string_view::size_type n = string_view::npos, char_type zero = '0', char_type one = '1')
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:549]: static_cast<char_type>('0'), static_cast<char_type>('1')

File: include/magic_enum_containers.hpp [hunk 6]
```diff
@@ -562,8 +562,8 @@ class bitset {
     }
   }
 
-  constexpr explicit bitset(detail::raw_access_t, const char* str, std::size_t n = ~std::size_t{}, char zero = '0', char one = '1')
-      : bitset(std::string_view{str, (std::min)(std::char_traits<char>::length(str), n)}, 0, n, zero, one) {}
+  constexpr explicit bitset(detail::raw_access_t, const char_type* str, std::size_t n = ~std::size_t{}, char_type zero = '0', char_type one = '1')
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:565]: static_cast<char_type>('0'), static_cast<char_type>('1')

File: include/magic_enum_containers.hpp [hunk 7]
```diff
@@ -562,8 +562,8 @@ class bitset {
...
+      : bitset(std::string_view{str, (std::min)(std::char_traits<char_type>::length(str), n)}, 0, n, zero, one) {}
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:566]: remove std from string_view

File: include/magic_enum_containers.hpp [hunk 8]
```diff
@@ -591,7 +591,7 @@ class bitset {
   }
 
   template <typename Cmp = std::equal_to<>>
-  constexpr explicit bitset(string_view sv, Cmp&& cmp = {}, char sep = '|') {
+  constexpr explicit bitset(string_view sv, Cmp&& cmp = {}, char_type sep = '|') {
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:594]: static_cast<char_type>('|')
Check this character too at static_assert check at magic_enum.hpp line 141/142

File: include/magic_enum_containers.hpp [hunk 9]
```diff
@@ -756,7 +756,7 @@ class bitset {
     return res;
   }
 
-  [[nodiscard]] string to_string(char sep = '|') const {
+  [[nodiscard]] string to_string(char_type sep = '|') const {
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:759]: static_cast<char_type>('|')

File: include/magic_enum_containers.hpp [hunk 10]
```diff
@@ -771,7 +771,7 @@ class bitset {
     return name;
   }
 
-  [[nodiscard]] string to_string(detail::raw_access_t, char zero = '0', char one = '1') const {
+  [[nodiscard]] string to_string(detail::raw_access_t, char_type zero = '0', char_type one = '1') const {
```
Comment (schaumb) [inline include/magic_enum_containers.hpp:774]: static_cast<char_type>('0'), static_cast<char_type>('1')

**Review by schaumb — COMMENTED** [review]
I think these default arguments need static_cast.

**Comment by Neargye** [comment]
Thanks for review!
Fixed in a separate commit

**Pull request merged**